src/surveys/components/ChoiceResponse.tsx

```diff
@@ -0,0 +1,85 @@
+import { VNode, h } from 'preact';
+import { useState } from 'preact/hooks';
+import { Button, Header } from '../../components';
+import { FormSettings, ID, Question, Theme } from '../../types';
+
+interface SingleResponseProps {
+  question: Question;
+  onAnswered: (questionId: ID, answerId: ID | ID[]) => void;
+  submitText?: string;
+  theme?: Theme;
+}
+
+export default function SelectResponse({
+  question,
+  onAnswered,
+  submitText,
+  theme,
+}: SingleResponseProps): VNode {
+  const [selectedOption, setSelectedOption] = useState<ID | ID[]>(
+    (question.settings as FormSettings).multiselect ? [] : ''
+  );
+
+  const handleOptionChange = (optionId: ID) => {
+    if ((selectedOption as ID[]).includes(optionId)) {
+      setSelectedOption((prevState) =>
+        (prevState as ID[]).filter((id) => id !== optionId)
+      );
+    } else {
+      setSelectedOption((prevState) => [...(prevState as ID[]), optionId]);
+    }
+  };
+
+  const handleSubmit = (e: h.JSX.TargetedEvent<HTMLFormElement, Event>) => {
+    e.preventDefault();
+    if (!selectedOption) return;
+
+    onAnswered(question.id, selectedOption);
+  };
+
+  return (
+    <form className={'max-w-sm space-y-4'} onSubmit={handleSubmit}>
+      <Header
+        title={question.label}
+        description={question.description}
+        color={theme?.question}
+      />
+
+      <div className={'space-y-2'}>
+        {(question.options ?? []).map((option) => (
```

Comment:
@afolasope I forgot to mention, in `MultipleSettings` or `SingleSettings`, there's `randomizeAnswer` flag. If set to true, it should randomise the option before rendering. Please don't overthink this, a basic shuffling technique should be enough. For example,

```
const shuffle = (options: QuestionOption[]) => {
  for (let i = 0; i < options.length; i++) {
    const j = Math.floor(Math.random() * (i + 1));
    [options[i], options[j]] = [options[j], options[i]];
  }
};

export const shuffleArray = (options: QuestionOption[], shuffleOption: string) => {
  const optionsCopy = [...options];

  if (shuffleOption === "all") {
    shuffle(optionsCopy);
  } else if (shuffleOption === "exceptLast") {
    const lastElement = optionsCopy.pop();
    shuffle(optionsCopy);
    optionsCopy.push(lastElement);
  }

  return optionsCopy;
};
```

Reply:
I have now added the `randomize` function